### DISCUSSION TITLE
Floating UI instead of Popper

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco.UI#301] [Ideas]
Currently it seems the UI library has a dependency on Popper.js which has been a popular tool for positioning tooltips, dropdowns etc.

I noticed [Shoelace](https://shoelace.style) library is using [Floating UI](https://floating-ui.com) states it's an evolution of Popper:
https://floating-ui.com/docs/motivation#comparison-with-popper-style-api

> [Popper](https://popper.js.org/) is currently the most popular open source solution to position floating elements, created in 2016. Prior to that it was [Tether](https://tetherjs.dev/docs/welcome/).

> Floating UI is the evolution of Popper v2, and aims to be a lower-level solution similar to CSS in which you progressively add properties to achieve desired positioning behavior.

**Comment by bjarnef** [comment]
@iOvergaard @nielslyngsoe something to consider. In case we want to use Floating UI instead, but probably not in v1 of UI library, here's a migration guide: https://floating-ui.com/docs/migration

**Reply by bjarnef** [reply]
@iOvergaard should we consider using Floating UI as replacement of Popper? Not sure how much UI library depends on Popper, but it seems it has some benefits https://floating-ui.com/docs/motivation

**Reply by iOvergaard** [reply]
Looks like a decent enough library to use. And looks like it's easier to handle edge cases with its programmatic approach to positioning. I would think we could make good use of it for the next version of UUI, so will definitely keep it on our radar.

**Reply by madsrasmussen** [reply]
There is a native popover API in the making. It might be worth looking into: https://developer.chrome.com/docs/web-platform/popover-api/

**Comment by bjarnef** [comment]
@iOvergaard @leekelleher 
I noticed TipTap v3 is in beta, and they have replaced `tippy.js` with `floating-ui`:
https://next.tiptap.dev/docs/resources/whats-new

However since I submitted this discussion it seems UI library doesn't have dependency on Popper.
